Mlem/Views/Tabs/Posts/Components/Post/Components/Community View/Community View.swift

```diff
@@ -102,12 +101,14 @@ struct CommunityView: View {
                                 account: account,
                                 appState: appState
                             )
-                        } catch let communityDetailsFetchingError {
-                            print("Failed while fetching community details: \(communityDetailsFetchingError)")
-
-                            appState.alertTitle = "Could not load community information"
-                            appState.alertMessage = "The server might be overloaded.\nTry again later."
-                            appState.isShowingAlert.toggle()
+                        } catch {
+                            print("Failed while fetching community details: \(error)")
+                            
+                            appState.contextualError = .init(
+                                title: "Could not load community information",
+                                message: "The server might be overloaded.\nTry again later.",
+                                underlyingError: error
```

Comment:
this is an example where we would want to communicate the error to the user, so we pass in some contextual information (title/message), the new error handling will look at the `underlyingError` first, if it's something wee have default handling for (like a session issue) it'll be handled, if not - an alert will be shown using the information we've fed in